tests/pos/splice-pat.scala

```diff
@@ -0,0 +1,21 @@
+import scala.quoted.*
+
+object MyMatcher {
+  def unapply(expr: Expr[Any])(using Quotes): Option[Expr[Int]] = ???
+}
+
+object MyMatcher2 {
+  def unapply(expr: Expr[Int])(using Quotes): Boolean = ???
+}
+
+def foo(x: Any): Unit = ???
+def bar(x: Int): Int = ???
+
+def oneLevel(expr: Expr[Any])(using Quotes): Expr[Int] = expr match
+  case '{ foo(${MyMatcher(y@MyMatcher2())}) } => y
+
+def twoLevel(expr: Expr[Any])(using Quotes): Expr[Int] = expr match
+  case '{ foo(${MyMatcher('{ bar(${y@MyMatcher2()}).getClass}) }) } => y
+
+def bindQuote(expr: Expr[Any])(using Quotes): Expr[Int] = expr match
+  case '{ foo(${y@'{bar($_)}})} => y
```

Comment:
You should also add a test in `tests/pos-macros` that will execute those quoted patterns when expanding some macro just to make sure that the contests of the splice are correctly desugared and compiled.

Reply:
Done

Reply:
Done